worker/provisioner/provisioner_task.go

```diff
@@ -1024,10 +1010,10 @@ func (task *provisionerTask) startMachine(
 	// increase if StartInstace failed with ErrAvailabilityZoneFailed
 	// until all availability zones have been tried.
 	for attemptsLeft := task.retryStartInstanceStrategy.retryCount; attemptsLeft >= 0; attemptsLeft-- {
-		if !placementZone {
+		if !strings.Contains(startInstanceParams.Placement, "zone") {
```

Comment:
The provisioner shouldn't need to know anything about placement, since we're got DeriveAvailabilityZone now

I think what you want is something like:

```go
// Iff the provider has chosen a zone, then AvailabilityZone will be non-empty.
distributeAcrossZones := startInstanceParams.AvailabilityZone == ""

for attemptsLeft := task.retryStartInstanceStrategy.retryCount; attemptsLeft >= 0; attemptsLeft-- {
    if distributeAcrossZones {
        ...
    }
    ...
}
```

Reply:
this was updated.